### PR TITLE
cake 5: Add native type hints for Log

### PR DESCRIPTION
**Pull request by othercorey**
@ADmad @markstory Also forgot to push Log.

### CODE REVIEW

File: src/Log/Log.php [hunk 1]
```diff
@@ -271,11 +272,11 @@ public static function levels(): array
      * ```
      *
      * @param array|string $key The name of the logger config, or an array of multiple configs.
-     * @param array|null $config An array of name => config data for adapter.
+     * @param \Psr\Log\LoggerInterface|\Closure|array|null $config An array of name => config data for adapter.
      * @return void
      * @throws \BadMethodCallException When trying to modify an existing config.
      */
-    public static function setConfig($key, $config = null): void
+    public static function setConfig(array|string $key, LoggerInterface|Closure|array|null $config = null): void
```
Comment (othercorey) [inline src/Log/Log.php:279]: I think `LoggerInterface` is the correct base class here. Wasn't sure if it had to be a `BaseLog` instance, but I don't think so.

**Comment by ADmad** [comment]
`psr/log` has new v3 release with types. Might be a good idea to update it too to avoid potentially having to change the types again.

**Review by markstory — APPROVED** [review]

**Comment by othercorey** [comment]
> `psr/log` has a new v3 release with types. Might be a good idea to update it too to avoid potentially having to change the types again.

I can only find a tag and no release notes. What's the difference between v2 and v3?

**Comment by ADmad** [comment]
> What's the difference between v2 and v3?

The addition of argument and return types (what you are doing currently :slightly_smiling_face:).

**Pull request merged**